armi/reactor/reactors.py

```diff
@@ -2129,6 +2129,48 @@ def growToFullCore(self, cs):
 
         return converter
 
+    def calculateNewGridPlatePitchFromTemp(self, cs, isoTemp: float) -> float:
+        """calculate a grid plate pitch using a new bulk ave core inlet temperature in C"""
+        # retrieve the first grid plate (all the same material, so just get the first)
+        gridPlateBlock = self.getFirstBlock(blockType=Flags.GRID_PLATE)
+        # retrieve common grid plate component, material, temp, and pitch
+        gridPlateComponent = gridPlateBlock.getComponent(Flags.GRID_PLATE)
+        gridPlateMaterial = gridPlateComponent.material
+        currentGridPlatePitch = self.getAssemblyPitch()
+        currentGridPlateTemp = cs["Tin"]
+        runLog.info(
+            f"Using a {Flags.GRID_PLATE} material of {gridPlateMaterial} for grid plate expansion/contraction calculations "
+            f"with an initial temperature of {currentGridPlateTemp} C and new temperature of {isoTemp} C."
+        )
+        # compute expansion/contraction fraction
+        dll = gridPlateMaterial.linearExpansionFactor(
+            Tc=isoTemp,
+            T0=currentGridPlateTemp,
+        )
+        # compute new grid place pitch
+        return currentGridPlatePitch * (1.0 + dll)
+
+    def updateGridPlatePitch(self, cs, isoTemp: float):
+        """update the grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        cs
+            case settings
+        isoTemp : float, optional
+            an isothermal temperature in C to apply to cs["Tin"]
+
+        Notes
+        -----
+        1) all grid plate blocks are the same material
+        2) all grid plate blocks are isothermal and equal to cs["Tin"]
+        """
+        newGridPlatePitch = self.calculateNewGridPlatePitchFromTemp(cs, isoTemp)
+        # update the pitch
+        self.setPitchUniform(newGridPlatePitch)
+        # update the bulk core average inlet temperature
+        cs["Tin"] = isoTemp
```

Comment:
It is perhaps not desirable to set Tin to this value. I could see a likely scenario when a user just wants to perturb the pitch to see the reactivity effect without messing with other calcs (i.e. TH)

Reply:
How about an optional parameter to update the inlet temp? Set the default to False.

Reply:
https://github.com/terrapower/armi/pull/993/commits/191f193736fd82a9ad91ae6e3d7080156c86dd08